plugins/module_utils/network/sonic/utils/bgp_utils.py

```diff
@@ -28,11 +28,195 @@
     'openconfig-bgp-types:IPV6_UNICAST': 'ipv6_unicast',
     'openconfig-bgp-types:L2VPN_EVPN': 'l2vpn_evpn',
 }
+
+AS_NOTATION_TYPES_MAP = {
+    'ASDOT': 'asdot',
+    'ASDOT_PLUS': 'asdot+',
+}
+
+AS_NOTATION_TO_TYPES_MAP = {
+    'asdot': 'ASDOT',
+    'asdot+': 'ASDOT_PLUS',
+}
+
 GET = "get"
 network_instance_path = '/data/openconfig-network-instance:network-instances/network-instance'
 protocol_bgp_path = 'protocols/protocol=BGP,bgp/bgp'
 
 
+def to_bgp_as_notation_request_type(as_notation):
+    """Convert as_notation types to Openconfig As-dot enums"""
+    return AS_NOTATION_TO_TYPES_MAP.get(as_notation)
+
+
+class BgpAsn(str):
+    """BgpAsn class to equate asdot+ and asplain"""
+    def __new__(cls, as_val):
+        if isinstance(as_val, str):
+            obj = super().__new__(cls, as_val)
+            obj.intval = int(as_val) if as_val.find('.') < 0 else (int(as_val.split('.')[0]) * 0x10000 + int(as_val.split('.')[1]))
+            return obj
+        if isinstance(as_val, int):
+            obj = super().__new__(cls, as_val)
+            obj.intval = as_val
+        else:
+            raise TypeError('Invalid BGP AS Number')
+        return obj
+
+    def __eq__(self, other):
+        if isinstance(other, BgpAsn):
+            return self.intval == other.intval
+        if isinstance(other, str):
+            if len(other) == 0:
+                return False
+            if other.find('.') < 0:
+                if other.isdigit():
+                    return self.intval == int(other)
+                return False
+            return self.intval == (int(other.split('.')[0]) * 0x10000 + int(other.split('.')[1]))
+        if isinstance(other, int):
+            return self.intval == other
+        return False
+
+    def __ne__(self, other):
+        return not self.__eq__(other)
+
+    def to_request(self):
+        """Return asn according to openconfig model (original input: asdot(+) as string, asplain as integer)"""
+        return self.intval if self.__str__().find('.') < 0 else self.__str__()
+
+
+def get_bgp_asn(cfglist):
```

Comment:
1, This function does not return value. So suggest to change the name to like transform_*** or convert_*** or something.
2, It handles both bgp and bgp_neighbors, so may I suggest to move it to individual config file to do the function specifically.

Reply:
1. Rename conversion functions to convert_xxx().
2. Other than the global sonic_bgp, the bgp_af, bgp_neighbors and bgp_neighbors_af have both global and neighbor configs that need to be converted.

Reply:
Thanks for comments.